server.go

```diff
@@ -108,7 +138,13 @@ func (s *Server) handlePutMhs(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusAccepted)
 }
 
-func (s *Server) handleGetMh(w http.ResponseWriter, r *http.Request) {
+func (s *Server) handleGetMh(w *responseWirterWithStatus, r *http.Request) {
+	start := time.Now()
+
+	defer func() {
+		s.m.RecordHttpLatency(context.Background(), time.Since(start).Milliseconds(), r.Method, r.URL.Path, w.status)
```

Comment:
Passing `r.URL.Path` straight up would result in explosive permutations of metrics on prometheus: we will endup with tags per looked up multihash here for example.

Instead of using `r.URL.Path` I recommend using a mapping to the path prefix. For example, `multihash` is enough as the value of `path` tag.

Ditto in other places.